deploy.sh

```diff
@@ -4,21 +4,36 @@ set -x
 
 VAULT_TOKEN=$1
 GIT_BRANCH=$2
+TARGET_ENV=$3
 
-#need to get the environment from the branch name
-if [ "$GIT_BRANCH" == "develop" ]; then
-	ENVIRONMENT="dev"
-elif [ "$GIT_BRANCH" == "alpha" ]; then
-    ENVIRONMENT="alpha"
-elif [ "$GIT_BRANCH" == "staging" ]; then
-	ENVIRONMENT="staging"
-elif [ "$GIT_BRANCH" == "master" ]; then
-	ENVIRONMENT="prod"
+set +x
+if [ -z "$TARGET_ENV" ]; then
+    echo "TARGET_ENV argument not supplied; inferring from GIT_BRANCH '$GIT_BRANCH'."
+
+    if [ "$GIT_BRANCH" == "develop" ]; then
+        TARGET_ENV="dev"
+    elif [ "$GIT_BRANCH" == "alpha" ]; then
+        TARGET_ENV="alpha"
+    elif [ "$GIT_BRANCH" == "staging" ]; then
+        TARGET_ENV="staging"
+    elif [ "$GIT_BRANCH" == "master" ]; then
+        TARGET_ENV="prod"
+    else
+        echo "Git branch '$GIT_BRANCH' is not configured to automatically deploy to a target environment"
+        exit 1
+    fi
+fi
+
+if [[ "$TARGET_ENV" =~ ^(dev|alpha|staging|prod)$ ]]; then
+    ENVIRONMENT=${TARGET_ENV}
 else
-	echo "Unknown Git branch $GIT_BRANCH"
-	#exit 1
+    echo "Unknown environment: $TARGET_ENV - must be one of [dev, alpha, staging, prod]"
```

Comment:
so we also have a perf deploy?

Reply:
@ahaessly not sure if you have a `perf` environment set up or not.  To add it, it is pretty easy, we just need to update your deploy script here, and then also update https://github.com/broadinstitute/dsp-jenkins/blob/gpolumbo-GAWB-3855/src/main/resources/IndieDeployConfig.conf to add the `perf` env as a viable target.